File: PortalApp.API/Models/Category.cs
using System;
using System.Collections.Generic;

namespace PortalApp.API.Models
{
    public class Category
    {
        public Guid? Id { get; set; }
        public string Title { get; set; }
        public string Type { get; set; }
        public string Icon { get; set; }
        public string Url { get; set; }
        public string TitleEng { get; set; }
        public string TitleKaz { get; set; }
        
        public ICollection<Category> Children { get; set; }
        public ICollection<CategoryProduct> CategoryProducts { get; set; }
    }
}

File: PortalApp-SPA/src/app/_models/category.ts
export class Category {
    id: string;
    title: string;
    type: string;
    icon: string;
    url: string;
    children?: Category[];
    expanded: boolean;
    selected: boolean;

    constructor(category?) {
        category = category || {};
        this.id = category.id || '';
        this.title = category.title || '';
        this.type = category.type || '';
        this.icon = category.icon || '';
        this.url = (category.url != null && category.url.length > 0) ? category.url : null;
        this.children = category.children || [];
        this.expanded = category.expanded || false;
        this.selected = category.selected || false;
    }
}
